apis/core/v1alpha1/commonrepository_types.go

```diff
@@ -0,0 +1,53 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// RepositoryStatus defines the observed state of a Repository.
+type RepositoryStatus struct {
+	// Conditions is a list of status conditions ths object is in.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// This field is not part of any API contract
+	// it will go away as soon as kubectl can print conditions!
+	// When evaluating object state in code, use .Conditions instead.
+	Phase RepositoryStatusPhase `json:"phase,omitempty"`
+	// Hash of image + config that was successfully unpacked.
+	UnpackedHash string `json:"unpackedHash,omitempty"`
+}
+
+// Repository condition types.
+const (
+	// A Repository "Available" condition tracks the availability of the underlying ObjectDeployment objects.
+	// When the Repository is reporting "Available" = True, it's expected that whatever the Repository installs
+	// is up and operational. Repository "Availability" may change multiple times during it's lifecycle.
+	RepositoryAvailable = "Available"
+	// Progressing indicates that a new release is being rolled out.
+	RepositoryProgressing = "Progressing"
```

Comment:
Repositories can't be "Progressing".

Reply:
Good catch @thetechnick. I removed it now 👍